src/lambda.rs

```diff
@@ -0,0 +1,212 @@
+    fn builder() -> LambdaResponseBuilder {
+        LambdaResponseBuilder {
+            status_code: 200,
+            status_description: None,
+            headers: HeaderMap::default(),
+            body: None,
+        }
+    }
+
+    fn from_message<M: ToString>(code: u16, message: M) -> Self {
+        Self::builder()
+            .status_code(code)
+            .header(X_RESTATE_SERVER, X_RESTATE_SERVER_VALUE)
+            .header(CONTENT_TYPE, "text/plain".parse().unwrap())
+            .body(Bytes::from(message.to_string()))
+            .build()
+    }
+}
+
+struct LambdaResponseBuilder {
+    status_code: u16,
+    status_description: Option<String>,
+    headers: HeaderMap,
+    body: Option<Base64Data>,
+}
+
+impl LambdaResponseBuilder {
+    pub fn status_code(mut self, status_code: u16) -> Self {
+        self.status_code = status_code;
+        self.status_description = http::StatusCode::from_u16(status_code)
+            .map(|s| s.to_string())
+            .ok();
+        self
+    }
+
+    pub fn header(mut self, key: HeaderName, value: HeaderValue) -> Self {
+        self.headers.insert(key, value.into());
+        self
+    }
+
+    pub fn body(mut self, body: Bytes) -> Self {
+        self.body = Some(Base64Data(body.into()));
+        self
+    }
+
+    pub fn build(self) -> LambdaResponse {
+        LambdaResponse {
+            status_code: self.status_code,
+            status_description: self.status_description,
+            headers: self.headers,
+            body: self.body,
+            is_base64_encoded: true,
+        }
+    }
+}
+
+impl From<Error> for LambdaResponse {
+    fn from(err: Error) -> Self {
+        LambdaResponse::from_message(err.status_code(), err.to_string())
+    }
+}
+
+/// Wraps an [`Endpoint`] to implement the `lambda_runtime::Service` trait for AWS Lambda.
+///
+/// This adapter allows a Restate endpoint to be deployed as an AWS Lambda function.
+/// It handles the conversion between Lambda's request/response format and the
+/// internal representation used by the SDK.
+#[derive(Clone)]
+pub struct LambdaEndpoint(Endpoint);
+
+impl LambdaEndpoint {
+    pub fn new(endpoint: Endpoint) -> Self {
+        Self(endpoint)
+    }
+
+    /// Runs the Lambda service.
+    ///
+    /// This function starts the `lambda_runtime` and begins processing incoming
+    /// Lambda events, passing them to the wrapped [`Endpoint`].
+    pub fn run(self) -> impl Future<Output = Result<(), lambda_runtime::Error>> {
+        let svc = service_fn(handle);
+        let svc = svc.map_request(move |req| {
+            let endpoint = self.0.clone();
+            LambdaEventWithEndpoint {
+                inner: req,
+                endpoint,
+            }
+        });
+
+        lambda_runtime::run(svc)
+    }
+}
+
+struct LambdaEventWithEndpoint {
+    inner: LambdaEvent<LambdaRequest>,
+    endpoint: Endpoint,
+}
+
+async fn handle(req: LambdaEventWithEndpoint) -> Result<LambdaResponse, Infallible> {
```

Comment:
Shall we use the instrument macro here as shown https://docs.aws.amazon.com/lambda/latest/dg/rust-logging.html#rust-logging-tracing?